sos/report/plugins/logs.py

```diff
@@ -59,13 +59,21 @@ def setup(self):
         self.add_cmd_output("journalctl --disk-usage")
         self.add_cmd_output('ls -alRh /var/log/')
 
-        journal = os.path.exists("/var/log/journal/")
-        if journal and self.is_installed("systemd"):
+        # collect journal logs if:
+        # - there is some data present, either persistent or runtime only
+        # - systemd-journald service exists
+        # otherwise fallback to collecting few well known logfiles directly
+        journal = os.path.exists("/var/log/journal/") or \
+            os.path.exists("/run/log/journal/")
+        if journal and self.is_service("systemd-journald"):
             self.add_journal(since=since)
             self.add_journal(boot="this", catalog=True)
             self.add_journal(boot="last", catalog=True)
             if self.get_option("all_logs"):
-                self.add_copy_spec("/var/log/journal/*")
+                self.add_copy_spec([
+                    "/var/log/journal/*",
+                    "/run/log/journal/*"
+                ])
```

Comment:
I guess I have two questions about this change: if there are files present at either location, what's the harm in just collecting them anyway? Are there valid situations where we *don't* want to do that (for space concerns, for e.g.)?

If would simplify things a bit if we could eliminate that logic.

I'm also a bit curious what the rationale is now for collecting raw journals vs. `journalctl` output; it seems if we have one maybe we don't need the other?

The block that we're discussing here (many commits ago) was originally intended to capture the journal logs on Fedora systems after Fedora moved ahead with disabling/removing syslog by default, so I'm not sure it's the best base for future work since the use of journals has become more pervasive and universal since this was first done.

Anyway; that falls a bit outside the scope of this specific patch review which looks generally fine to me outside the formatting nitpicks.